src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -69,6 +69,9 @@ public void EnlistInDistributedTransactionIfNeeded(ISessionImplementor session)
 						}
 						session.ConnectionManager.AfterTransaction();
 						session.AfterTransactionCompletion(wasSuccessful, null);
+						foreach (var dependentSession in session.ConnectionManager.DependentSessions)
+							dependentSession.AfterTransactionCompletion(wasSuccessful, null);
+
```

Comment:
`ConnectionManager` could do the job too. It already is responsible for a part of transaction handling.